server/routes/updateGoal/updateGoalController.ts

```diff
@@ -36,7 +36,7 @@ export default class UpdateGoalController {
     req.session.updateGoalForm = undefined
 
     const goalCreatedDate = moment(updateGoalForm.createdAt)
-    const goalTargetCompletionDate = moment(updateGoalForm.targetCompletionDate)
+    const goalTargetCompletionDate = moment(updateGoalForm.originalTargetCompletionDate)
```

Comment:
Here we now use the `originalTargetCompletionDate` from the form (which never gets mutated) to create the radio button option, rather than `targetCompletionDate` (which will be `Invalid date` if the user has tried to submit an invalid date)